gateway/proxy_eth.go

```diff
@@ -353,13 +353,13 @@ func (gw *Node) EthMaxPriorityFeePerGas(ctx context.Context) (ethtypes.EthBigInt
 	return gw.target.EthMaxPriorityFeePerGas(ctx)
 }
 
-func (gw *Node) EthEstimateGas(ctx context.Context, tx ethtypes.EthCall) (ethtypes.EthUint64, error) {
+func (gw *Node) EthEstimateGas(ctx context.Context, tx ethtypes.EthCall, blkParam ethtypes.EthBlockNumberOrHash) (ethtypes.EthUint64, error) {
```

Comment:
Does this still work if the user _doesn't_ pass `blkParam`? IIRC, this is why, e.g., methods like `EthFeeHistory` take a `jsonrpc.RawParams`. That lets one decode a variable number of params.

E.g., see https://github.com/filecoin-project/lotus/blob/3283cb70a776d5fba26658b38e3e9f000e9b6f3f/node/impl/full/eth.go#L692

https://github.com/filecoin-project/lotus/blob/3283cb70a776d5fba26658b38e3e9f000e9b6f3f/chain/types/ethtypes/eth_types.go#L803-L835

Reply:
It requires passing `blkParam` so yeah, this will probably break other tools since now the expect two params. I will update the PR tomorrow morning to support this as optional param (works with 1 and 2 params)